### PR TITLE
fix to replace incorrect tag names retrieved from api meta model

### LINKED ISSUE #192: boot_device should be device
## The problem

The name for boot_device in XMLBootDeviceWriteMany should be device for the tag name, not boot_device

## Environment

* Desktop OS/version
* Java version to run maven
* Go version under test
* The metamodel version speicifed in the root pom.xml
* The model version sepecified in the root pom.xml

## Description

[Description of the issue]

## Steps to Reproduce

1. [First Step]
2. [Second Step]
3. [and so on...]

**Expected behavior:** [What you expect to happen]

**Actual behavior:** [What actually happens]

**Reproduces how often:** [What percentage of the time does it reproduce?]

## Link to logs [ Good To Have ]

Create a [GIST](https://gist.github.com) which is a paste of your _full_ logs, and link them here.

### PR DESCRIPTION
**Pull request by mathianasj**
### Description of the Change

This allows replacing a tag name if the api model does not provide the correct xml tag name based on the class name.

### Possible Drawbacks

We are creating a band-aid instead of having the library providing the name be fixed

### Verification Process

Ran a generation and used the library to interface with ovirt

### Applicable Issues

closes #192

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by mwperina** [comment]
Ori, could you please take a look? How was this solved in Ruby and Python SDKs?

@oliel

**Comment by mathianasj** [comment]
@imjoey I liked how the solution was presented in the reference you gave.  I went ahead and altered my request to use that method so we would not need to maintain the list of replacements in this project.  I also added two unit tests to ensure that it works with one and two boot devices.

File: generator/src/main/java/org/ovirt/sdk/go/WritersGenerator.java [hunk 1]
```diff
@@ -252,8 +252,14 @@ else if (memberType instanceof StructType || memberType instanceof EnumType) {
         else if (memberType instanceof ListType) {
             ListType listType = (ListType) memberType;
             Type elementType = listType.getElementType();
-            String elementSingularTag = goNames.getTagStyleName(elementType.getName());
             if (elementType instanceof StructType || elementType instanceof EnumType) {
+                String elementSingularTag;
+                if(elementType instanceof StructType) {
+                    elementSingularTag = goNames.getTagStyleName(elementType.getName());
+                } else {
+                    elementSingularTag = schemaNames.getSchemaTagName(names.getSingular(member.getName()));
```
Comment (imjoey) [inline generator/src/main/java/org/ovirt/sdk/go/WritersGenerator.java]:
```suggestion
                    elementSingularTag = goNames.getTagStyleName(names.getSingular(memberName));
```

Hi @mathianasj , one more tiny changes to keep code consistency. The function `goNames.getTagStyleName()` simply wraps the `schemaNames.getSchemaTagName()`, and `memberName` is already  defined before.
Thank you so much.

**Comment by imjoey** [comment]
> Ori, could you please take a look? How was this solved in Ruby and Python SDKs?
> 
> @oliel

Hi @mwperina and @oliel , I ran the generators for Python SDK and Ruby SDK and ensured that they are not affected by this. The tag generated for enum `BootDevice` is `device` as expected. After digging into the codes of the generators, you could find the codes at [1] for Python and [2] for Ruby, which exactly do the right thing.
Please let me know if I'm wrong. Thanks.

[1]: https://github.com/oVirt/ovirt-engine-sdk/blob/master/generator/src/main/java/org/ovirt/sdk/python/WritersGenerator.java#L307
[2]: https://github.com/oVirt/ovirt-engine-sdk-ruby/blob/master/generator/src/main/java/org/ovirt/sdk/ruby/WritersGenerator.java#L262

**Comment by oliel** [comment]
> > Ori, could you please take a look? How was this solved in Ruby and Python SDKs?
> > @oliel
> 
> Hi @mwperina and @oliel , I ran the generators for Python SDK and Ruby SDK and ensured that they are not affected by this. The tag generated for enum `BootDevice` is `device` as expected. After digging into the codes of the generators, you could find the codes at [1](https://github.com/oVirt/ovirt-engine-sdk/blob/master/generator/src/main/java/org/ovirt/sdk/python/WritersGenerator.java#L307) for Python and [2](https://github.com/oVirt/ovirt-engine-sdk-ruby/blob/master/generator/src/main/java/org/ovirt/sdk/ruby/WritersGenerator.java#L262) for Ruby, which exactly do the right thing.
> Please let me know if I'm wrong. Thanks.

I'll have time to take a deeper look only on Monday, but it looks correct to me

File: generator/src/main/java/org/ovirt/sdk/go/WritersGenerator.java (hunk 1 above)
Comment (mathianasj) [reply generator/src/main/java/org/ovirt/sdk/go/WritersGenerator.java]: @imjoey made these changes from suggestion

**Comment by imjoey** [comment]
Hi @mathianasj , let's merge it. Thanks very much for your contribution.

**Comment by mathianasj** [comment]
Will this through some form of ci/cd be generated and propagated to the generated repo?

**Comment by imjoey** [comment]
> Will this through some form of ci/cd be generated and propagated to the generated repo?

@mathianasj  yep, the oVirt/go-ovirt repo will be updated once the TravisCI succeed. Since this is a big bugfix, I'll also make a tag for this repo to trigger a new version release oVirt/go-ovirt then we can update the dependency of oVirt/go-ovirt.

**Pull request merged**